File: server/datasets/java_source-code/Multiple-C1/organic_6.java
/*
 * Graph BFS - Organic Submission #6
 * Uses integer-indexed nodes (0..n-1) and a boolean visited array.
 */

import java.util.*;

class Solution {
    public static List<Integer> bfs(int numNodes, int[][] adjacencyList, int start) {
        boolean[] visited = new boolean[numNodes];
        List<Integer> order = new ArrayList<>();
        Queue<Integer> q = new ArrayDeque<>();

        visited[start] = true;
        q.add(start);

        while (!q.isEmpty()) {
            int node = q.poll();
            order.add(node);
            for (int nxt : adjacencyList[node]) {
                if (!visited[nxt]) {
                    visited[nxt] = true;
                    q.add(nxt);
                }
            }
        }
        return order;
    }

    public static void main(String[] args) {
        int n = 6;
        int[][] adj = {
            {1, 2},
            {0, 3, 4},
            {0, 5},
            {1},
            {1, 5},
            {2, 4}
        };
        System.out.println(bfs(n, adj, 0));
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_c1/organic_6.py
"""
Graph BFS - Organic Submission #6
Uses integer-indexed nodes (0..n-1) and a boolean visited array.
"""

from collections import deque


def bfs(num_nodes, adjacency_list, start):
    visited = [False] * num_nodes
    order = []
    q = deque()

    visited[start] = True
    q.append(start)

    while q:
        node = q.popleft()
        order.append(node)
        for nxt in adjacency_list[node]:
            if not visited[nxt]:
                visited[nxt] = True
                q.append(nxt)

    return order


if __name__ == "__main__":
    n = 6
    adj = [
        [1, 2],     # 0
        [0, 3, 4],  # 1
        [0, 5],     # 2
        [1],        # 3
        [1, 5],     # 4
        [2, 4],     # 5
    ]
    print(bfs(n, adj, 0))
